frontend/src/queries/zarrQueries.ts

```diff
@@ -43,6 +44,26 @@ export function isZarrDirectory(files: FileOrFolder[] | undefined): boolean {
   return files.some(file => zarrFileNames.includes(file.name));
 }
 
+/**
+ * Detects which Zarr versions are supported by checking for version-specific marker files.
+ * @returns Array of supported versions: ['v2'], ['v3'], or ['v2', 'v3']
+ */
+export function detectZarrVersions(fileNames: string[]): ('v2' | 'v3')[] {
```

Comment:
This is nice. Can it also replace uses of `isZarrDirectory`, which looks like it now duplicates a subset of this logic?